homeassistant-supervised/usr/sbin/hassio-supervisor

```diff
@@ -1,37 +1,105 @@
+if [ -z "${SUPERVISOR_IMAGE_ID}" ]; then
+    # Get the latest from update information
+    # Using updater information instead of config. If the config version is
+    # broken, this creates a way (e.g., bad release).
+    SUPERVISOR_VERSION=$(jq -r '.supervisor // "latest"' "${SUPERVISOR_DATA}/updater.json" || echo "latest")
+
+    echo "[WARNING] Supervisor image missing, downloading a fresh one: ${SUPERVISOR_VERSION}"
 
-runSupervisor() {
-    docker rm --force hassio_supervisor || true
+    # Pull in the Supervisor
+    if docker pull "${SUPERVISOR_IMAGE}:${SUPERVISOR_VERSION}"; then
+        # Tag as latest if versioned
+        if [ "${SUPERVISOR_VERSION}" != "latest" ]; then
+            docker tag "${SUPERVISOR_IMAGE}:${SUPERVISOR_VERSION}" "${SUPERVISOR_IMAGE}:latest"
+        fi
+    else
+        # Pull failed, updater info might be corrupted, re-trying with latest
+        echo "[WARNING] Supervisor downloading failed trying: latest"
+        docker pull "${SUPERVISOR_IMAGE}:latest"
+    fi
 
+    SUPERVISOR_IMAGE_ID=$(docker inspect --format='{{.Id}}' "${SUPERVISOR_IMAGE}" || echo "")
+fi
+
+if [ -n "${SUPERVISOR_CONTAINER_ID}" ]; then
+    # Image changed, remove previous container
+    if [ "${SUPERVISOR_IMAGE_ID}" != "${SUPERVISOR_CONTAINER_ID}" ]; then
+        echo "[INFO] Supervisor image has been updated, destroying previous container..."
+        docker container rm --force hassio_supervisor || true
+        SUPERVISOR_CONTAINER_ID=""
+    fi
+
+    # Start script changed, remove previous container
+    # shellcheck disable=SC3013
+    if [ ! -f "${SUPERVISOR_STARTSCRIPT_VERSION}" ] || [ "${SUPERVISOR_STARTSCRIPT_VERSION}" -nt "$0" ] || [ "${SUPERVISOR_STARTSCRIPT_VERSION}" -ot "$0" ]; then
+        echo "[INFO] Supervisor start script has changed, destroying previous container..."
+        docker container rm --force hassio_supervisor || true
+        SUPERVISOR_CONTAINER_ID=""
+    fi
+fi
+
+# If Supervisor container is missing, create it
+if [ -z "${SUPERVISOR_CONTAINER_ID}" ]; then
+    echo "[INFO] Creating a new Supervisor container..."
     # shellcheck disable=SC2086
-    docker run --name hassio_supervisor \
-        --privileged \
-        --security-opt apparmor=hassio-supervisor \
-        --security-opt seccomp=unconfined \
+    docker container create \
+        --name hassio_supervisor \
+        --privileged --security-opt apparmor="hassio-supervisor" \
         -v /run/docker.sock:/run/docker.sock:rw \
         -v /run/dbus:/run/dbus:ro \
+        -v /run/supervisor:/run/os:rw \
         -v /run/udev:/run/udev:ro \
         -v /etc/machine-id:/etc/machine-id:ro \
-        -v "${HASSIO_DATA}":/data:rw \
-        -e SUPERVISOR_SHARE="${HASSIO_DATA}" \
+        -v ${SUPERVISOR_DATA}:/data:rw \
+        -e SUPERVISOR_SHARE=${SUPERVISOR_DATA} \
         -e SUPERVISOR_NAME=hassio_supervisor \
-        -e SUPERVISOR_MACHINE="${MACHINE}" \
-        "${SUPERVISOR}"
-}
+        -e SUPERVISOR_CPU_RT=1 \
```

Comment:
Not sure if that works on Debian as it needs a kernel module for this

Reply:
I guess we can just remove `SUPERVISOR_CPU_RT` for now